glide-core/src/client/value_conversion.rs

```diff
@@ -67,16 +67,100 @@ pub(crate) fn convert_to_expected_type(
             Value::Nil => Ok(value),
             _ => Ok(Value::Double(from_owned_redis_value::<f64>(value)?.into())),
         },
+        ExpectedReturnType::ArrayOfDouble => match value {
+            Value::Array(array) => {
+                let result: Result<Vec<Value>, _> = array
+                    .into_iter()
+                    .map(|element| match element {
+                        Value::BulkString(_) => Ok(Value::Double(
+                            from_owned_redis_value::<f64>(element)?.into(),
+                        )),
+                        _ => Ok(element),
+                    })
+                    .collect();
+
+                result.map(Value::Array)
+            }
+            _ => Err((
+                ErrorKind::TypeError,
+                "Response couldn't be converted to Array",
+                format!("(response was {:?})", value),
+            )
+                .into()),
+        },
     }
 }
 
+fn convert_array_to_map(
```

Comment:
this can be much simpler:

```rs
fn convert_array_to_map(
    array: Vec<Value>,
    expected_key_return_type: Option<ExpectedReturnType>,
    expected_value_return_type: Option<ExpectedReturnType>,
) -> RedisResult<Value> {
    let mut map = Vec::with_capacity(array.len() / 2);
    let mut iterator = array.into_iter();
    while let Some(key) = iterator.next() {
        let Some(value) = iterator.next() else {
            return Err((
                ErrorKind::TypeError,
                "Response has odd number of items, and cannot be entered into a map",
            )
                .into());
        };
        map.push((
            convert_to_expected_type(key, expected_key_return_type)?,
            convert_to_expected_type(value, expected_value_return_type)?,
        ));
    }

    Ok(Value::Map(map))
}
```

you'll just need to add `#[derive(Clone, Copy)]` to `ExpectedReturnType`.